tests/src/Kernel/Plugin/Field/FieldFormatter/DateAndTextFormatterKernelTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\Tests\apigee_m10n\Kernel\Plugin\Field\FieldFormatter;
+
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\entity_test\Entity\EntityTest;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\Tests\apigee_m10n\Kernel\MonetizationKernelTestBase;
+
+/**
+ * Test the `apigee_date_and_text_formatter` field formatter.
+ *
+ * @group apigee_m10n
+ * @group apigee_m10n_kernel
+ */
+class DateAndTextFormatterKernelTest extends MonetizationKernelTestBase {
+
+  /**
+   * The formatter manager.
+   *
+   * @var \Drupal\Core\Field\FormatterPluginManager
+   */
+  protected $formatter_manager;
+
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $field_manager;
+
+  /**
+   * The field's display settings.
+   *
+   * @var array
+   */
+  protected $settings;
+
+  /**
+   * Extra modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
+    'system',
+    'field',
+    'text',
+    'entity_test',
+    'user',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installConfig(['system']);
+    $this->installConfig(['field']);
+    $this->installEntitySchema('entity_test');
+
+    $this->entityType = 'entity_test';
+    $this->bundle = $this->entityType;
+    $this->fieldName = mb_strtolower($this->randomMachineName());
+
+    $field_storage = FieldStorageConfig::create([
+      'field_name' => $this->fieldName,
+      'entity_type' => $this->entityType,
+      'type' => 'timestamp',
+    ]);
+    $field_storage->save();
+
+    $instance = FieldConfig::create([
+      'field_storage' => $field_storage,
+      'bundle' => $this->bundle,
+      'label' => $this->randomMachineName(),
+    ]);
+    $instance->save();
+
+    $this->display = entity_get_display($this->entityType, $this->bundle, 'default')
+      ->setComponent($this->fieldName, [
+        'type' => 'boolean',
```

Comment:
We are just setting the default display for the field to a very simple formatter, so that we later switch it to the actual formatter during the test. Core does the same in the [timestamp formatter test](https://git.drupalcode.org/project/drupal/blob/8.8.x/core/modules/field/tests/src/Kernel/Timestamp/TimestampFormatterTest.php#L73-78).

Reply:
Got it, would be good to point that out in a comment to be more clear